### ISSUE TITLE
Cannot perform inference with ApproxLP2 and IntervalFactor

### ISSUE THREAD

**Issue by cbonesana** [IDSIA/crema#56]
As title says, when one declares a `DAGModel<IntervalFactor>` and he try to run an inference using `ApproxLP2`, no models work.

**Comment by rcabanasdepaz** [comment]
When we have intervals such that for any of the dimensions the lower and upper bounds are the same, some problems might occur due to the numerical stability of double/float numbers. This is solved by defining epsilon value close to 0.0 for separating those bounds. Here you have an example. I guess your problem could be related to this. Do you have the code where you get some problems?

Here you have a working example.


```
        double eps = 0.000000001; // eps > 0. If set to 0.0, the inference will not work.

        DAGModel model = new DAGModel();
        int x = model.addVariable(2);
        int u = model.addVariable(3);
        model.addParent(x,u);

        BayesianFactor ifx = new BayesianFactor(model.getDomain(x,u));
        ifx.setData(new double[] {
                1., 0.,
                1., 0.,
                0., 1.,
        });
        model.setFactor(x, ifx);

        IntervalFactor ifu = new IntervalFactor(model.getDomain(u), model.getDomain());
        ifu.set(new double[] { 0, 0, 0.8-eps}, new double[] { 0.2, 0.2, 0.8 });
        model.setFactor(u, ifu);

        Inference inference = new Inference();
        double[] upper = inference.query(model, x).getUpper();
        System.out.println(Arrays.toString(upper));  // [0.20000001, 0.8]

        ApproxLP2 approxLP2 = new ApproxLP2();
        upper = approxLP2.query(model, x).getUpper();
        System.out.println(Arrays.toString(upper));  // [0.20000001, 0.8]


```

**Comment by cbonesana** [comment]
No @rcabanasdepaz , that's not the issue I found. With @davidhuber we found an issue with how the `IntervalFactor`s are managed to be converted for the solver.

This issue causes always a `NotFeasibleException` if the variables are not declared in a particular order. The correct behavior is that for any order of declaration of the variables, the converter builds the correct linear problem for the solver. This is currently not true.

I apologize for not been too much specific, bun I'm not an expert in this field and I @davidhuber is already working on a fix for that.
